### PR TITLE
Refactored StaticFile and StaticFolder - moved type to folder #3737

### PR DESCRIPTION
**Pull request by PritishC**
Note: nereid needs to be released for the build to pass.

### CODE REVIEW

File: static_file.py [hunk 1]
```diff
@@ -123,7 +124,7 @@ def get_post_form_args(self):
         connection to S3 via Boto and returns a dictionary, which can then be
         processed on the client side.
         """
-        if not self.folder.s3_use_bucket:
+        if not self.folder.type == 's3':
```
Comment (priyankarani) [inline static_file.py]: self.folder.type != 's3' is better

File: static_file.py [hunk 2]
```diff
@@ -108,14 +111,12 @@ def check_cloudfront_cname(self):
 class NereidStaticFile:
     __name__ = "nereid.static.file"
 
-    is_s3_bucket = fields.Function(
-        fields.Boolean("S3 Bucket?"), 'get_is_s3_bucket'
-    )
```
Comment (priyankarani) [inline static_file.py]: I think this function field should be there to check if file belongs to s3.

you can modify getter to check if folder type is s3

File: static_file.py [hunk 3]
```diff
@@ -28,22 +28,21 @@ class NereidStaticFolder:
     __name__ = "nereid.static.folder"
     _rec_name = "folder_name"
 
-    s3_use_bucket = fields.Boolean("Use S3 Bucket?")
     s3_access_key = fields.Char(
         "S3 Access Key",
-        states={'required': Bool(Eval('s3_use_bucket'))}
+        states={'required': Bool(Eval('type') == 's3')}
```
Comment (priyankarani) [inline static_file.py]: it should be invisible too if not s3

File: static_file.py (hunk 2 above)
Comment (PritishC) [inline static_file.py]: As discussed with @prakashpp, we won't be keeping this field.

File: static_file.py (hunk 1 above)
Comment (PritishC) [inline static_file.py]: Done.

File: static_file.py (hunk 3 above)
Comment (PritishC) [inline static_file.py]: Done.

File: static_file.py [hunk 4]
```diff
@@ -108,14 +103,12 @@ def check_cloudfront_cname(self):
...
 
-    is_s3_bucket = fields.Function(
-        fields.Boolean("S3 Bucket?"), 'get_is_s3_bucket'
-    )
     s3_key = fields.Function(
         fields.Char("S3 key"), getter="get_s3_key",
         searcher="search_s3_key"
     )
     is_large_file = fields.Boolean('Is Large File')
+    is_private = fields.Boolean("Is Private")
```
Comment (priyankarani) [inline static_file.py]: can you put a question mark '?' after Is Private

File: tests/test_nereid_s3.py [hunk 5]
```diff
@@ -29,6 +29,12 @@
     test_depends
 from trytond.transaction import Transaction
 from trytond.exceptions import UserError
+from trytond.config import config
+
+config.add_section('nereid_s3')
```
Comment (priyankarani) [inline tests/test_nereid_s3.py]: this should be added in setup.py

File: static_file.py [hunk 6]
```diff
@@ -28,22 +29,9 @@ class NereidStaticFolder:
...
-    s3_bucket_name = fields.Char(
-        "S3 Bucket Name",
-        states={'required': Bool(Eval('s3_use_bucket'))}
-    )
     s3_cloudfront_cname = fields.Char(
         "S3 Cloudfront CNAME",
-        states={'required': Bool(Eval('s3_use_bucket'))}
+        states={'required': Bool(Eval('type') == 's3')}
     )
     s3_object_prefix = fields.Char("S3 Object Prefix")
```
Comment (priyankarani) [inline static_file.py]: Can you add states to this ? to make it visible only for s3

Comment (PritishC) [inline static_file.py]: Not sure if states need to be added, as we have these fields in a group in the XML, and the group is set to invisible if type is not s3.

File: static_file.py (hunk 4 above)
Comment (PritishC) [inline static_file.py]: Done.

File: tests/test_nereid_s3.py (hunk 5 above)
Comment (PritishC) [inline tests/test_nereid_s3.py]: Done.

File: static_file.py (hunk 6 above)
Comment (priyankarani) [inline static_file.py]: still its recommended to add states in python code too

**Pull request merged**